internal/inspect/policy.go

```diff
@@ -50,37 +52,76 @@ func (pol *Policy) Inspect(p *policyv1.Policy) error {
 
 	policyID := namer.PolicyKey(p)
 	localVariables := policy.ListVariables(p)
+	referencedAttributes, referencedDerivedRoles, referencedVariables, err := pol.inspectDefinitionsAndRules(p)
+	if err != nil {
+		return fmt.Errorf("failed to inspect definitions and rules of the policy %s: %w", policyID, err)
+	}
+
 	if _, ok := p.PolicyType.(*policyv1.Policy_ExportVariables); ok {
 		sortedLocalVariables := make([]*responsev1.InspectPoliciesResponse_Variable, 0, len(localVariables))
 		for _, lv := range localVariables {
 			sortedLocalVariables = append(sortedLocalVariables, lv)
 		}
 		sort.Slice(sortedLocalVariables, func(i, j int) bool {
-			return sortedLocalVariables[i].Name < sortedLocalVariables[j].Name
+			return sort.StringsAreSorted([]string{sortedLocalVariables[i].Name, sortedLocalVariables[j].Name})
 		})
 
+		attributes := make([]*responsev1.InspectPoliciesResponse_Attribute, 0, len(referencedAttributes))
+		for _, attr := range referencedAttributes {
+			attributes = append(attributes, &responsev1.InspectPoliciesResponse_Attribute{
+				Name: attr.Name,
+				Type: attr.Type,
+			})
+		}
+		if len(attributes) > 0 {
+			sort.Slice(attributes, func(i, j int) bool {
+				return sort.StringsAreSorted([]string{attributes[i].Name, attributes[j].Name})
+			})
+		}
+
 		if len(localVariables) > 0 {
 			pol.results[policyID] = &responsev1.InspectPoliciesResponse_Result{
-				Variables: sortedLocalVariables,
-				PolicyId:  storeIdentifier,
+				Attributes: attributes,
+				Variables:  sortedLocalVariables,
+				PolicyId:   storeIdentifier,
 			}
 		}
 
 		return nil
 	}
 
-	referencedDerivedRoles, referencedVariables, err := pol.inspectDefinitionsAndRules(p)
+	referencedAttributesFromVariables, err := pol.inspectVariableDefinitions(p)
```

Comment:
Why can't this be done as part of the `inspectDefinitionsAndRules` function above?